### PR TITLE
Work

### PR DESCRIPTION
**Pull request by JosePaumard**
Updates on Sealed Classes, Switch Statements and Pattern Matching

### CODE REVIEW

File: step/records.md [hunk 1]
```diff
@@ -498,6 +524,7 @@ For more details on Records, please check the following resources.
 * [JEP 395: Records](https://openjdk.java.net/jeps/395)
 * [Java Feature Spotlight: Records](https://inside.java/2020/02/04/spotlightrecords/)
 * [Simpler Serialization with Records](https://inside.java/2021/03/12/simpler-serilization-with-records/)
+* [JEP Café Episod #01](https://youtu.be/l1VrmvyIEpM)
```
Comment (delabassee) [inline step/records.md]: typo -> Episode

File: step/sealed-classes.md [hunk 2]
```diff
@@ -27,14 +27,14 @@ Additionally a sealed class imposes three constraints on its permitted subclasse
...
-    * A permitted subclass may be declared final to prevent its part of the class hierarchy from being extended further.
+    * A permitted subclass may be declared final to prevent its part of the class hierarchy from being extended further. It can also be implicitly final (enumerated types and records).
     * A permitted subclass may be declared sealed to allow its part of the hierarchy to be extended further than envisaged by its sealed superclass, but in a restricted fashion.
     * A permitted subclass may be declared non-sealed so that its part of the hierarchy reverts to being open for extension by unknown subclasses. (A sealed class cannot prevent its permitted subclasses from doing this.)
  
 
 ## Your first Sealed Classes
 
-💡 **Sealed Classes** is a preview feature in JDK 16 so make sure that preview features are enabled both at compile time and run-time.
+💡 **Sealed Classes** is a feature in JDK 17.
```
Comment (delabassee) [inline step/sealed-classes.md]: we tend to say 'standard and permanent feature'

File: step/sealed-classes.md [hunk 3]
```diff
@@ -344,20 +344,12 @@ It can be accessed via `{public_ip}:8080/sessions`.
...
-
-
-
-
-
+In this exercise, you have used **Sealed Classes**. Sealed Classes is a new feature that enables a developer to define a restricted classes hierarchy, i.e. a developer has now the ability to explicitly states for a given class (or an interface) which classes (or interfaces) may extend (or implement) it. Sealed Classes is a feature in JDK 17.
 
+For more details, please check [JEP 409: Sealed Classes](https://openjdk.java.net/jeps/409) and the following [Java Feature Spotlight: Sealed Classes](https://www.infoq.com/articles/java-sealed-classes/) article.
 
+You can also check the [JEP Café Episod #02](https://youtu.be/652kheEraHQ).
```
Comment (delabassee) [inline step/sealed-classes.md]: Episod**e**

File: step/switch-expression.md [hunk 4]
```diff
@@ -178,6 +178,8 @@ var trackDetail = switch (speaker.track()) {
 
 In this exercise, you have used **Switch Expressions**, a standard feature since Java 14.
 
-Switch expressions complement nicely the traditional Swith statement by enabling to easily write less error-prone Switch cases. Moreover, the produced code is also more readable! For additional details, please check [JEP 361: Switch Expressions (Standard)](https://openjdk.java.net/jeps/361).
+Switch expressions complement nicely the traditional Swith statement by enabling to easily write less error-prone Switch cases. Moreover, the produced code is also more readable!
```
Comment (delabassee) [inline step/switch-expression.md]: typo: raditional **Swith** statement

File: step/switch-expression.md [hunk 5]
```diff
@@ -178,6 +178,8 @@ var trackDetail = switch (speaker.track()) {
 
 In this exercise, you have used **Switch Expressions**, a standard feature since Java 14.
 
-Switch expressions complement nicely the traditional Swith statement by enabling to easily write less error-prone Switch cases. Moreover, the produced code is also more readable! For additional details, please check [JEP 361: Switch Expressions (Standard)](https://openjdk.java.net/jeps/361).
+Switch expressions complement nicely the traditional Swith statement by enabling to easily write less error-prone Switch cases. Moreover, the produced code is also more readable! 
 
+For additional details, please check [JEP 361: Switch Expressions (Standard)](https://openjdk.java.net/jeps/361).
 
+You can also check the [JEP Café Episod #03](https://youtu.be/652kheEraHQ).
```
Comment (delabassee) [inline step/switch-expression.md]: Episod**e** ;-)

**Review by delabassee — APPROVED** [review]

File: step/sealed-classes.md (hunk 2 above)
Comment (JosePaumard) [reply step/sealed-classes.md]: Done

File: step/switch-expression.md (hunk 4 above)
Comment (JosePaumard) [reply step/switch-expression.md]: Done

File: step/switch-expression.md (hunk 5 above)
Comment (JosePaumard) [reply step/switch-expression.md]: Done

File: step/records.md (hunk 1 above)
Comment (JosePaumard) [reply step/records.md]: Done

File: step/sealed-classes.md (hunk 3 above)
Comment (JosePaumard) [reply step/sealed-classes.md]: Done

**Pull request merged**